### PR TITLE
Reproject 0.9 changes

### PR DESCRIPTION
**Pull request by nabobalis**
I set the old defaults to suppress the warnings, I am unsure if this is a good idea.

Q: Do you want to bump reprojects version?
A: NO MUST WORK WITH 0.8

### CODE REVIEW

File: ndcube/ndcube.py [hunk 1]
```diff
@@ -640,7 +640,7 @@ def reproject_to(self, target_wcs, algorithm='interpolation', shape_out=None, or
             algorithm is selected).
             For 'interpolation' algorithm, this can be any of: 'nearest-neighbor', 'bilinear',
             'biquadratic', and 'bicubic'.
-            For 'adaptive' algorithm, this can be either 'nearest-neighbor' or 'bilinear'.
+            For 'adaptive' algorithm, this has no affect.
```
Comment (nabobalis) [inline ndcube/ndcube.py:643]:
```suggestion
            For 'adaptive' algorithm, this has no effect.
```

File: ndcube/ndcube.py [hunk 2]
```diff
@@ -709,8 +709,13 @@ def reproject_to(self, target_wcs, algorithm='interpolation', shape_out=None, or
                                     return_footprint=return_footprint)
 
         elif algorithm == 'adaptive':
+            if order:
+                warnings.warn(
+                    "Order is ignored for the adaptive algorithm",
+                    UserWarning
+                )
```
Comment (nabobalis) [inline ndcube/ndcube.py:716]: Maybe its better to not raise a warning and just let it be ignored?

File: ndcube/ndcube.py [hunk 3]
```diff
@@ -709,8 +709,13 @@ def reproject_to(self, target_wcs, algorithm='interpolation', shape_out=None, or
...
             data = reproject_adaptive(self, output_projection=target_wcs, shape_out=shape_out,
-                                      order=order, return_footprint=return_footprint)
+                                      return_footprint=return_footprint, kernel="Gaussian", boundary_mode="ignore")
```
Comment (nabobalis) [inline ndcube/ndcube.py:718]: Extra defaults to suppress warnings from reproject.

**Comment by Cadair** [comment]
Two comments also made in chat:

* Need to maintain compatibility with <0.9
* We should align map and ndcube (by changes in either).

I will look at this more tomorrow.

File: ndcube/ndcube.py [hunk 4]
```diff
@@ -614,64 +614,103 @@ def _validate_algorithm_and_order(self, algorithm, order):
             raise ValueError(f"The 'algorithm' argument must be one of "
                              f"{', '.join(order_compatibility.keys())}.")
 
-    def reproject_to(self, target_wcs, algorithm='interpolation', shape_out=None, order='bilinear',
-                     output_array=None, parallel=False, return_footprint=False):
+    def reproject_to(self, target_wcs, algorithm='interpolation', return_footprint=False, **reproject_args):
```
Comment (nabobalis) [inline ndcube/ndcube.py:617]: I here you like breaking API.

File: ndcube/ndcube.py [hunk 5]
```diff
@@ -694,7 +733,16 @@ def reproject_to(self, target_wcs, algorithm='interpolation', shape_out=None, or
...
             raise ValueError('Given target_wcs is not compatible with this NDCube, the physical types do not match.')
 
-        # If shape_out is not specified explicity, try to extract it from the low level WCS
+        # LOL AT THIS
+        shape_out = reproject_args.get("shape_out", None)
+        order = reproject_args.get("order", 'bilinear')
+        output_array = reproject_args.get("output_array", None)
+        parallel = reproject_args.get("parallel", False)
+        reproject_args.get("kernel", "Gaussian")
+        reproject_args.get("boundary_mode", "ignore")
```
Comment (nabobalis) [inline ndcube/ndcube.py:742]: This is one way I guess?

**Comment by nabobalis** [comment]
Doc build is fixed in another PR.

File: ndcube/ndcube.py (hunk 5 above)
Comment (nabobalis) [reply ndcube/ndcube.py:742]: I can probably pop these, pass the final `reproject_args` to reproject instead?

**Comment by Cadair** [comment]
@MeeseeksDev hello

**Comment by Cadair** [comment]
I am going to adopt this PR.

**Pull request closed without merging**